packages/webdriver-utils/test/providers/automateProvider.test.js

```diff
@@ -148,48 +148,48 @@ describe('AutomateProvider', () => {
       const automateProvider = new AutomateProvider('1234', 'https://localhost/command-executor', { platform: 'win' }, {}, {}, 'client', 'environment', {}, percyBuildInfo);
       await automateProvider.createDriver();
-      automateProvider.driver.executeScript = jasmine.createSpy().and.rejectWith(new Error('Random network error'));
+      spyOn(Driver.prototype, 'executeScript').and.returnValue(Promise.reject(new Error('Random network error')));
       await expectAsync(automateProvider.percyScreenshotBegin('abc')).toBeRejectedWithError('Random network error');
     });
 
     it('marks the percy session as success', async () => {
       const automateProvider = new AutomateProvider('1234', 'https://localhost/command-executor', { platform: 'win' }, {}, {}, 'client', 'environment', {}, percyBuildInfo);
       await automateProvider.createDriver();
-      automateProvider.driver.executeScript = jasmine.createSpy().and.returnValue(Promise.resolve({ success: true }));
+      spyOn(Driver.prototype, 'executeScript').and.returnValue(Promise.resolve({ success: true }));
       await automateProvider.percyScreenshotBegin('abc');
       expect(automateProvider._markedPercy).toBeTruthy();
     });
 
     it('throw error if statusCode:13', async () => {
       const automateProvider = new AutomateProvider('1234', 'https://localhost/command-executor', { platform: 'win' }, {}, {}, 'client', 'environment', {}, percyBuildInfo);
       await automateProvider.createDriver();
-      automateProvider.driver.executeScript = jasmine.createSpy().and.returnValue(Promise.resolve({ status: 13, value: 'OS/Browser/Selenium combination is not supported' }));
+      spyOn(Driver.prototype, 'executeScript').and.returnValue(Promise.resolve({ status: 13, value: 'OS/Browser/Selenium combination is not supported' }));
       await expectAsync(automateProvider.percyScreenshotBegin('abc')).toBeRejectedWithError('OS/Browser/Selenium combination is not supported');
     });
 
     it('throw "Got invalid error resposne" if result.value does not exists', async () => {
       const automateProvider = new AutomateProvider('1234', 'https://localhost/command-executor', { platform: 'win' }, {}, {}, 'client', 'environment', {}, percyBuildInfo);
       await automateProvider.createDriver();
-      automateProvider.driver.executeScript = jasmine.createSpy().and.returnValue(Promise.resolve({ status: 13 }));
+      spyOn(Driver.prototype, 'executeScript').and.returnValue(Promise.resolve({ status: 13 }));
       await expectAsync(automateProvider.percyScreenshotBegin('abc')).toBeRejectedWithError('Got invalid error response');
     });
 
     it('mark percy sesssion as failure', async () => {
       const automateProvider = new AutomateProvider('1234', 'https://localhost/command-executor', { platform: 'win' }, {}, {}, 'client', 'environment', {}, percyBuildInfo);
       await automateProvider.createDriver();
       // eslint-disable-next-line prefer-promise-reject-errors
```

Comment:
do we still need to disable eslint here?

Reply:
While it's advised to create a new Error object, the error we encounter here is a response error rather than a function call error. Consequently, we find it necessary to temporarily disable linting in this context.